engine/debugHook.mli

```diff
@@ -0,0 +1,32 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+(* registration of debugger hooks *)
+type debugger_action =
+  | DbStep
+  | DbSkip
+  | DbExit
+  | DbHelp
+  | DbRunCnt of int
+  | DbRunBreakpoint of string
+  | DbFailure
+
+type debugger_hooks = {
+  read_cmd : unit -> debugger_action;   (* read a debugger command from the client *)
+  print_notice : Pp.t -> unit;          (* print a notice *)
+  print_debug  : Pp.t -> unit;          (* print a debug *)
+  print_prompt : Pp.t -> unit           (* print the debugger prompt *)
+}
```

Comment:
Maybe we could structure the responses here a bit more? I'd suggest you add a type `debugger_response` and let the backends print it; I'm surprised that the debugger would for example output with different "levels".

Reply:
In coqtop, "debug" gives you this lovely purple tag:

![image](https://user-images.githubusercontent.com/1253341/109760789-27f2ab00-7ba4-11eb-9310-59b76d128fed.png)